pkg/volume/csi/csi_plugin.go

```diff
@@ -892,35 +891,28 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 		return nil, errors.New(log("CSI driver reporting empty array for supported versions"))
 	}
 
-	// Sort by lowest to highest version
-	sort.Slice(versions, func(i, j int) bool {
-		parsedVersionI, err := utilversion.ParseGeneric(versions[i])
-		if err != nil {
-			// Push bad values to the bottom
-			return true
-		}
-
-		parsedVersionJ, err := utilversion.ParseGeneric(versions[j])
-		if err != nil {
-			// Push bad values to the bottom
-			return false
-		}
-
-		return parsedVersionI.LessThan(parsedVersionJ)
-	})
-
+	var highestSupportedVersion *utilversion.Version
+	var theErr error
 	for i := len(versions) - 1; i >= 0; i-- {
-		highestSupportedVersion, err := utilversion.ParseGeneric(versions[i])
+		currentHighestVer, err := utilversion.ParseGeneric(versions[i])
 		if err != nil {
-			return nil, err
+			theErr = err
+			continue
 		}
-
-		if highestSupportedVersion.Major() <= 1 {
-			return highestSupportedVersion, nil
+		if currentHighestVer.Major() > 1 {
+			continue
+		}
+		if highestSupportedVersion == nil {
+			highestSupportedVersion = currentHighestVer
+		} else if highestSupportedVersion.LessThan(currentHighestVer) {
+			highestSupportedVersion = currentHighestVer
 		}
 	}
 
-	return nil, errors.New(log("None of the CSI versions reported by this driver are supported"))
+	if highestSupportedVersion != nil {
+		return highestSupportedVersion, nil
+	}
+	return nil, errors.New(fmt.Sprintf("None of the CSI versions reported by this driver are supported : %v", theErr))
```

Comment:
nit: can you also add emit `versions` so that it is clear what the driver returned that Kubernetes didn't like..